### ISSUE TITLE
graphMaps in POMs have no effect

### ISSUE THREAD

**Issue by ArenasGuerreroJulian** [ontop/ontop#390]
Hi!

I am trying ontop v4.0.3 and I have the following R2RML mapping:

```
<agency>
	rr:logicalTable [
		rr:tableName "AGENCY";
	];
	rr:subjectMap [
		rr:template "http://example.org/agency/{\"agency_id\"}";
		rr:termType <http://www.w3.org/ns/r2rml#IRI>;
		rr:class <http://example.org/terms#agency>;
		rr:graphMap [ rr:constant <http://example.org#graph_1> ];
	];
	rr:predicateObjectMap [
		rr:predicateMap [ rr:constant <http://example.org#has_url> ];
		rr:graphMap [ rr:constant <http://example.org#graph_2> ];
		rr:objectMap[
			rr:column "agency_url";
		];
	];
.
```

The graphMap in the POM have no effect and any triples with `<http://example.org#graph_2>` is generated.

**Comment by ArenasGuerreroJulian** [comment]
(Forgot to mention!) I am using the materialization mode :)

**Comment by bcogrel** [comment]
Hi Julian,

Thanks for reporting and bringing the precision about the materialization mode, it is likely an important detail.

I think some refactoring has already been done in that direction some months ago (e.g. not using a CONSTRUCT query for the materialization as it cannot produce quads). 

Could you please give a try to the version4 branch? If you use Maven, you can give a try to the snapshot version. Alternatively, you can build the CLI https://ontop-vkg.org/dev/build.html .

Best,
Benjamin

**Comment by ArenasGuerreroJulian** [comment]
Hi Benjamin!

I have tried version4 branch and it generates the correct result :)

```
<http://example.org/agency/CRTM> <http://www.w3.org/1999/02/22-rdf-syntax-ns#type> <http://example.org/terms#agency> <http://example.org#graph_1> .
<http://example.org/agency/CRTM> <http://example.org#has_url> "http://www.crtm.es" <http://example.org#graph_1> .
<http://example.org/agency/CRTM> <http://example.org#has_url> "http://www.crtm.es" <http://example.org#graph_2> .
```

Best,
Julián

**Comment by kontchakov** [comment]
Hi Julián

It was fixed with 58f9897f5a1fb2fae98bb0ee0e816ab0d36d60c4 (Nov 2020) and will be included in the next official release.

Best
Roman
